### ISSUE TITLE
Trident install to vanilla Kubernetes cluster fails if oc tool is present

### ISSUE THREAD

**Issue by stobias123** [NetApp/trident#213]
If the `oc` tool is present, but you're targeting a kubernetes cluster, `tridentctl` will try to install openshift resources.

```
$ ./tridentctl install -n trident
WARN Could not delete installer cluster role binding.  error="error: unable to recognize \"STDIN\": no matches for kind \"ClusterRoleBinding\" in version \"authorization.openshift.io/v1\"\n; exit status 1"
WARN Could not delete installer cluster role.      error="error: unable to recognize \"STDIN\": no matches for kind \"ClusterRole\" in version \"authorization.openshift.io/v1\"\n; exit status 1"
WARN Could not modify security context constraint.  error="Error from server (NotFound): the server could not find the requested resource (get securitycontextconstraints.security.openshift.io privileged)\n; exit status 1"
```

**Comment by clintonk** [comment]
True, we did make this simplifying assumption.  If **oc** is available, we assume it's an OpenShift environment.  Would you please tell us more about what you're trying to do?

**Comment by stobias123** [comment]
Not trying to do anything - I run vanilla kubernetes sometimes, and sometimes I run openshift. 

Noticed the bug and thought I should report, took me like a 25mins to figure out what was going on... (did I clean up correctly, is there some magic cache, is my kubeconfig screwed up, etc.)

While it may be an edge case, I figured I'd save others the pain.

**Comment by innergy** [comment]
It would be useful to be able to override this assumption.

**Comment by scaleoutsean** [comment]
You could load a custom (`oc`-less $PATH) during installation.

**Comment by gnarl** [comment]
This issue is fixed with [commit 2f0563f](https://github.com/NetApp/trident/commit/2f0563f29a5b4e5a013b27c32132f09969acfb40) which is included with the Trident 21.10 release.
